encode.js

```diff
@@ -0,0 +1,42 @@
+var keyEncoding = 'utf8';
+var valueEncoding = 'binary';
+
+module.exports = {
+  encodeKey: encodeKey,
+  encodeValue: encodeValue,
+  decodeValue: decodeValue,
+  keyEncoding: keyEncoding,
+  valueEncoding: valueEncoding,
+};
+
+function encodeKey(input) {
+  if (Buffer.isBuffer(input)) {
+    return input.toString(keyEncoding);
+  } else {
+    return input;
+  }
+}
+
+function encodeValue(input) {
+  if (Buffer.isBuffer(input)) {
+    input = { type: 'Buffer', data: input.toString(valueEncoding, valueEncoding) };
+  } else {
+    input = String(input)
+  }
+  input = JSON.stringify(input);
+  return input;
+}
+
+function decodeValue(input, asBuffer) {
+  var input = JSON.parse(input);
+  if (input && input.type === 'Buffer' && input.data) {
```

Comment:
not the case. binary just provides more efficient encoding. things come in as either buffers or values, and we need to 'recode' one of them in order to know how to process what we get out.
I opted to recode buffers instead of recoding values. One thing I wasn't sure about is when we were given a buffer to store and asBuffer is false, what do we return?